testing/trino-tests/src/test/java/io/trino/security/TestFunctionsInViewsWithFileBasedSystemAccessControl.java

```diff
@@ -0,0 +1,159 @@
+        queryRunner.createCatalog("mock", "mock");
+        return queryRunner;
+    }
+
+    @Test
+    public void testPtfSecurityDefinerViewCreatedByAlice()
+    {
+        assertQuerySucceeds(ALICE_USER, "CREATE VIEW blackhole.default.view_ptf_alice_security_definer SECURITY DEFINER AS SELECT * FROM TABLE(mock.system.simple_table_function())");
+        String securityDefinerQuery = "SELECT * FROM blackhole.default.view_ptf_alice_security_definer";
+        assertQuerySucceeds(ALICE_USER, securityDefinerQuery);
+        assertQuerySucceeds(BOB_USER, securityDefinerQuery);
+        assertQuerySucceeds(CHARLIE_USER, securityDefinerQuery);
+    }
+
+    @Test
+    public void testPtfSecurityInvokerViewCreatedByAlice()
+    {
+        assertQuerySucceeds(ALICE_USER, "CREATE VIEW blackhole.default.view_ptf_alice_security_invoker SECURITY INVOKER AS SELECT * FROM TABLE(mock.system.simple_table_function())");
+        String securityInvokerQuery = "SELECT * FROM blackhole.default.view_ptf_alice_security_invoker";
+        assertQuerySucceeds(ALICE_USER, securityInvokerQuery);
+        assertQuerySucceeds(BOB_USER, securityInvokerQuery);
+        assertAccessDenied(CHARLIE_USER, securityInvokerQuery, "Cannot access catalog mock");
+    }
+
+    @Test
+    public void testFunctionSecurityDefinerViewCreatedByAlice()
+    {
+        assertQuerySucceeds(ALICE_USER, "CREATE VIEW blackhole.default.view_function_alice_security_definer SECURITY DEFINER AS SELECT now() AS t");
+        String securityDefinerQuery = "SELECT * FROM blackhole.default.view_function_alice_security_definer";
+        assertQuerySucceeds(ALICE_USER, securityDefinerQuery);
+        assertQuerySucceeds(BOB_USER, securityDefinerQuery);
+        assertQuerySucceeds(CHARLIE_USER, securityDefinerQuery);
+    }
+
+    @Test
+    public void testFunctionSecurityInvokerViewCreatedByAlice()
+    {
+        assertQuerySucceeds(ALICE_USER, "CREATE VIEW blackhole.default.view_function_alice_security_invoker SECURITY INVOKER AS SELECT now() AS t");
+        String securityInvokerQuery = "SELECT * FROM blackhole.default.view_function_alice_security_invoker";
+        assertQuerySucceeds(ALICE_USER, securityInvokerQuery);
+        assertQuerySucceeds(BOB_USER, securityInvokerQuery);
+        assertAccessDenied(CHARLIE_USER, securityInvokerQuery, "Cannot execute function now");
+    }
+
+    @Test
+    public void testPtfSecurityDefinerViewCreatedByBob()
+    {
+        assertQuerySucceeds(BOB_USER, "CREATE VIEW blackhole.default.view_ptf_bob_security_definer SECURITY DEFINER AS SELECT * FROM TABLE(mock.system.simple_table_function())");
+        String securityDefinerQuery = "SELECT * FROM blackhole.default.view_ptf_bob_security_definer";
+        assertAccessDenied(ALICE_USER, securityDefinerQuery, "View owner does not have sufficient privileges: 'bob' cannot grant 'mock\\.system\\.simple_table_function' execution to user 'alice'");
+        assertQuerySucceeds(BOB_USER, securityDefinerQuery);
```

Comment:
That's an interesting behaviour, that `Bob` himself can actually query the view but no one else can. It's probably because view owner = view invoker.

Reply:
`io.trino.sql.analyzer.StatementAnalyzer.Visitor#analyzeView`